Documentation/specs/runtime-configuration-file.md

```diff
@@ -25,8 +25,15 @@ The files are both JSON files stored in UTF-8 encoding. Below are sample files.
 ```json
 {
     "runtimeOptions": {
-        "gcServer": true,
-        "gcConcurrent": false,
+
+        "configProperties": {
+            "System.GC.Server": true,
```

Comment:
in full framework, there is a ton of random `COMPlus_` config switches (https://github.com/dotnet/coreclr/blob/master/src/inc/clrconfigvalues.h). These config switches can be controlled via config file as well. There is desire to bring some of these config switches to .NET Core.

The problem with what we have is:
- The names are very random (e.g. `gcConcurrent`, `Thread_UseAllCpuGroups`). My suggestion was to use the `Namespace.Name` to give it some reasonable naming convention.
- The boundaries between runtime and framework are blurry. E.g. Is threadpool runtime or framework? In CoreCLR, it is in coreclr.dll (-> runtime). In CoreRT, it is in separate .dll (-> framework). My suggestion was to not to differentiate between the two.